crates/claims/crates/vc/src/syntax/non_empty_vec.rs

```diff
@@ -0,0 +1,90 @@
+use serde::{Deserialize, Serialize};
+use std::ops::Deref;
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+#[serde(try_from = "Vec<T>", into = "Vec<T>")]
+pub struct NonEmptyVec<T: Clone>(Vec<T>);
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
```

Comment:
If the error type has only one variant, you can just define an unit struct `EmptyVec` or `EmptyVecError`.